src/libpgagroal/utils.c

```diff
@@ -797,13 +817,17 @@ pgagroal_set_connection_proc_title(int argc, char** argv, struct connection* con
 {
    struct configuration* config;
    int primary;
-   char info[MISC_LENGTH];
+   char info[MAX_PROCESS_TITLE_LENGTH];
 
    config = (struct configuration*)shmem;
+   if (pgagroal_get_primary(&primary))
+   {
+      // cannot find the primary, this is a problem!
+      pgagroal_set_proc_title(argc, argv, connection->username, connection->database);
+      return;
+   }
 
-   pgagroal_get_primary(&primary);
-
-   snprintf(info, MISC_LENGTH, "%s@%s:%d",
+   snprintf(info, MAX_PROCESS_TITLE_LENGTH, "%s@%s:%d",
```

Comment:
To be sure that the string is `\0` terminated it should be

```
   snprintf(info, MAX_PROCESS_TITLE_LENGTH - 1, "%s@%s:%d",
```

Reply:
Fixed.